pkgs/applications/editors/ne/default.nix

```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchurl, ncurses, texinfo, tetex, perl, ghostscript }:
+
+
+stdenv.mkDerivation {
+  name = "ne-3.0.1";
+  version = "3.0.1";
+  src = fetchurl {
+    url = https://github.com/vigna/ne/archive/3.0.1.tar.gz;
+    md5 = "ec3dc82d10c271e95ecde630538f9f19";
+  };
+  buildInputs = [ ncurses tetex texinfo perl ghostscript ];
+  preBuild = ''
+    substituteInPlace src/makefile --replace "CC=c99" "cc=gcc"
+    substituteInPlace src/makefile --replace "-lcurses" "-lncurses"
+    substituteInPlace makefile --replace "./version.pl" "perl version.pl"
+    cd doc && make && cd ..
```

Comment:
This is the problem, it doesn't build with just make because it doesn't call the makefile in the doc folder and the one in the src folder, then fails when installing because files are missing. Also, if I just let the genericBuilder run `make PREFIX=$out install` it doesn't throw errors but the files are not there after it's done for some reason, while if I write the command myself it works.

I'm making some changes so that the code is a little cleaner, but I'm afraid I couldn't fix everything. You are welcome to try, but I think part of the fault is due to ne's problematic makefiles and I'm not really capable to fix them much.